src/quo2/components/token_tag.cljs

```diff
@@ -0,0 +1,68 @@
+(ns quo2.components.token-tag
+  (:require [quo2.foundations.colors :as colors]
+            [quo.react-native :as rn]
+            [quo.theme :as theme]
+            [status-im.ui.components.icons.icons :as icons]
+            [quo2.components.text :as text]))
+
+(def themes {:light   {:background-color colors/neutral-20}
+             :dark  {:background-color colors/neutral-80}})
+
+(defn get-value-from-size [size, big-option, small-option]
+  (if  (= size :big) big-option small-option))
+
+(defn style-container [size border-color is-required]
+  (merge {:height             32
+          :min-width         (get-value-from-size size 94 76)
+          :align-items        :center
+          :flex-direction     :row
+          :left             0
+          :border-radius      100
+          :padding-right      10}
+         (if is-required {:border-color     border-color
+                          :border-width     1} {})))
+
+(defn token-tag
+  "[token-tag opts \"label\"]
+   opts
+   {
+    :token string
+    :value string
+    :size :small/:big
+    :icon-name :icon-name
+    :border-color :color
+    :is-required true/false
+    :is-purchasable true/false
+    }"
+  [_ _]
+  (fn [{:keys [token value size icon-name border-color is-required is-purchasable] :or {size :small}}]
+    [rn/view {:style   (merge (style-container size border-color is-required)  (get-in themes [(theme/get-theme)]))}
+
+     [rn/view {:style {:margin-left 2
+                       :margin-right  (get-value-from-size size 8 6)}}
+      [icons/icon icon-name {:no-color true
+                             :size    24}]]
+
+     [text/text  {:weight          :medium
+                  :number-of-lines 1
+                  :size (get-value-from-size size :paragraph-2 :label)
+                  :padding 2
+                  :padding-top 12
+                  :margin-right (get-value-from-size size  12 10)} value " " token]
+     (when (or is-required is-purchasable) [rn/view {:style {:display :flex
+                                                             :align-items :center
+                                                             :justify-content :center
+                                                             :position :absolute
+                                                             :background-color  (if is-required border-color colors/neutral-50)
+                                                             :border-radius      100
+                                                             :border-color     (if (=  (theme/get-theme) :dark) colors/black colors/white)
+                                                             :border-width     1
+                                                             :right     -20
```

Comment:
perhaps there's a better way to get the purchase and checkmark icon sitting in the right place as per the designs 🤔